src/components/Preview/Preview.tsx

```diff
@@ -1,48 +1,47 @@
 import * as React from 'react'
 import { Loader } from 'decentraland-ui'
 
-import { EditorWindow } from './Preview.types'
+import { EditorWindow, Props, State } from './Preview.types'
 import './Preview.css'
 
 const editorWindow = window as EditorWindow
 
-export default class Preview extends React.Component {
-  private canvas = React.createRef<HTMLDivElement>()
+export default class Preview extends React.Component<Props, State> {
+  canvas = React.createRef<HTMLDivElement>()
 
   componentDidMount() {
-    // We're adding this to the end of the stack to allow the browser to paint the rest of the DOM first
-    setTimeout(() => this.startEditor(), 0)
-  }
-
-  shouldComponentUpdate() {
-    // We don't want to lose the reference to the canvas
-    return false
+    this.startEditor().catch(error => console.error('Failed to start editor', error))
   }
 
   async startEditor() {
-    if (!editorWindow.isDCLInitialized && editorWindow.initDCL) {
-      editorWindow.initDCL()
-    }
+    await editorWindow.editor.initEngine()
+
     try {
       const canvas = await editorWindow.editor.getDCLCanvas()
-      this.moveCanvas(canvas)
+
+      if (this.canvas.current && canvas) {
+        this.setState({
+          isLoading: false
+        })
+        this.canvas.current!.appendChild(canvas)
+        editorWindow.editor.resize()
+        this.props.onOpenEditor()
+      }
     } catch (error) {
       console.error('Failed to load Preview', error)
     }
   }
 
-  moveCanvas = (canvas: HTMLCanvasElement) => {
-    if (this.canvas.current && canvas) {
-      this.canvas.current.appendChild(canvas)
-      editorWindow.editor.initEngine()
-      editorWindow.editor.resize()
-    }
-  }
-
   render() {
+    const { isLoading } = this.props
```

Comment:
I'm not sure I understand the Props/State for this Component. If we're getting isLoading as prop, why are we updating the state on `startEditor`?

Reply:
due to the enablement of dani's chot mode